src/RA_StringUtils.h

```diff
@@ -466,22 +464,7 @@ class StringBuilder
 
     struct PendingString
     {
-        union {
-            const std::string* String;
-            const std::wstring* WString;
-
-            struct
-            {
-                const char* Pointer;
-                size_t Length;
-            } Char;
-
-            struct
-            {
-                const wchar_t* Pointer;
-                size_t Length;
-            } WChar;
-        } Ref{};
+        std::variant<const std::string*, const std::wstring*, gsl::cstring_span<>, gsl::cwstring_span<>> Ref;
```

Comment:
`cstring_span `and `cwstring_span` seemed like good candiates since they have pointer and size information.